sockeye/beam_search.py

```diff
@@ -604,6 +609,11 @@ def forward(self,
         eos_dist = mx.nd.full((batch_size * self.beam_size, self.output_vocab_size), val=np.inf,
                               ctx=self.context, dtype=self.dtype)
         eos_dist[:, C.EOS_ID] = 0
+        unk_dist = None
+        if self.prevent_unk:
+            unk_dist = mx.nd.full((batch_size * self.beam_size, self.output_vocab_size), val=0,
```

Comment:
you could simplify `mx.nd.zeros_like(eos_dist)`, context is already taken from the argument of `zeros_like()`

Reply:
`mx_nd.zeros_like` seems cleaner but it causes pylint error 
```
E1137: 'unk_dist' does not support item assignment (unsupported-assignment-operation)
```
because of the assignment `unk_dist[:, C.UNK_ID] = np.inf`. Is there any solution for this?

Reply:
Pylint isn't always right. You can try to add a type annotation in line 612: `unk_dist = None. # type: Optional[mx.nd.NDArray]` and see if that fixes it. Otherwise you could ignore the specific pylint error in that line.

Reply:
add `unk_dist = None # type: Optional[mx.nd.NDArray]` but it doesn't help. Ignoring pylint error for now

Reply:
it seems that I have to manually add `pylint --rcfile=pylintrc test -E --disable= E1137` to the code

Reply:
you add a comment to line 614: `#. pylint: ignore: E1137` or something like this.